src/rate_limiter/src/lib.rs

```diff
@@ -497,6 +621,15 @@ mod tests {
         }
     }
 
+    // After a restore, we cannot be certain that the last_update field has the same value.
+    fn equal_token_buckets(restored_tb: &TokenBucket, tb: &TokenBucket) -> bool {
```

Comment:
nit: I'd move this inside `impl TokenBucket` with `&self` standing for `tb` and the other parameter being the restored one (or vice versa, doesn't matter as long as the convention is made obvious by naming or comments)

Reply:
Yes, it would look better like Alexandra said.
FYI: to do that, you can add methods to `impl Whatever {}` inside `mod tests {}` too (as long as Whatever is in the current crate).

Since this is only tests you can even impl PartialEq and then just use `==`.
```rust
mod tests {
    impl PartialEq for TokenBucket {
        fn eq(&self, other: &Self) -> bool {...}
    }
}
```

Reply:
I moved the method to `impl TokenBucket`. TokenBucket is already deriving `PartialEq` and this led to conflicts when trying to derive it in the `tests` module.

Reply:
Yeah `TokenBucket` derives `PartialEq` because `RateLimiter` implements `PartialEq` and relies on comparing exact matches of `TokenBuckets`... This would be a *more* partial Eq but Rust doesn't have anything more flexible in terms of equivalence (the `Eq` trait exists but relies on `PartialEq` for comparison and only exists in a separate form to guarantee equivalence). Maybe you could get away with a `PartialOrd` in the tests module but it's hacky and doesn't really represent an ordering relationship. So I think it's better to leave the fn in `impl TokenBucket`